### PR TITLE
Added Advdupe2 warning error for invalid class args

### PR DESCRIPTION
**Pull request by thegrb93**

### CODE REVIEW

**Comment by TomyLobo** [comment]
I have no idea what's going on. can you split that up into two commits, where one refactors the code to use symbolic constants and the other commit does the actual change?

**Comment by thegrb93** [comment]
I'd rather not. Not much here has changed; just read the left side and understand what it does and then the right side. Have open the TYPE enum documentation so you know what the numbers from before represent.

**Comment by thegrb93** [comment]
I think I'll add an error notification to the FilterEntityTable too.

**Comment by TomyLobo** [comment]
Ok let me list everything that I see in there *in addition to* switching to symbolic constants:
- reversing an if
    - why? you're increasing indentation depth, making the code slightly less readable, for no apparent gain
- turning a blacklist for entity/function into a whitelist for bool, number, vector, angle and string
    - why?
- moved a "local" declaration
    - Are you sure this isn't a deoptimization? Also: why?
- and THEN you added the ErrorNoHalt

Really this is way too much for a single commit.
So many changes, some even controversial, rolled into one commit, that's bad style and hard to follow.

**Comment by thegrb93** [comment]
It's cleaner. It's a whitelist now because anything else will not be serialized. Yes and because it's cleaner.
